draft-ietf-quic-multipath.md

```diff
@@ -1487,68 +1487,65 @@ TBD-13 (experiments use 0x004e4f5f4349445f) | NO_CID_AVAILABLE | Path abandoned
 on source address spoofing as outlined in {{Section 21.5.4 of QUIC-TRANSPORT}} equally apply.
 Similarly, the anti-amplification limits as specified in {{Section 8 of QUIC-TRANSPORT}} need to be
 followed to limit the amplification risk.
 
-However, while {{QUIC-TRANSPORT}} only allows the use of one path simultaneously
-and therefore only one path migration at the time should be validated,
-this extension allows for multiple open paths, that could in theory be migrated
-all at the same time, and it allows for multiple paths that could be initialized
-simultaneously. Therefore, each path could be used to further amplify an attack.
-Endpoints need to limit the number of maximum paths and might consider
+{{QUIC-TRANSPORT}} only allows the use of one path
+and as such only one path validation at a time.
+This extension, however, allows for multiple open paths that could in theory be migrated
+all at the same time. Further multiple paths could be initialized
+simultaneously. Each open path could be used to further amplify an attack.
+Therefore, endpoints need to limit the maximum number of paths and might consider
 additional measures to limit the number of concurrent path validation processes
 e.g. by pacing them out or limiting the number of path initiation attempts
 over a certain time period.
 
 
-## Use of Transport Layer Security and the AEAD Encryption Nonce
+## Cryptographic handshake and AEAD Nonce
 
 The multipath extension as specified in this document is only enabled after a
 successful handshake when both endpoints indicate support for this extension.
 Respectively, all new frames defined in this extension are only used in 1-RTT packets.
 As the handshake is not changed by this extension, the transport security mechanisms
 as specified in {{QUIC-TLS}}, such as encryption key exchange and peer authentication,
-remain unchanged as well and the respective security considerations in {{QUIC-TLS}} applied unaltered.
-Note that with the use of this extension, multiple nonces can be in use simultaneously
-for the same AEAD key.
-
-Further note, that the limits as discussed on Appendix B of {{QUIC-TLS}}
+remain unchanged. As such, the respective security considerations in {{QUIC-TLS}} apply unaltered. However, note that the limits as discussed on Appendix B of {{QUIC-TLS}}
 apply to the total number of packets sent on all paths.
 
 This specification changes the AEAD calculation by using the path identifier as part of
-AEAD encryption nonce (see {{multipath-aead}}). To ensure a unique nonce, path identifiers
-are limited to 32 bits and cannot be reused for another path in the same connection.
-
+AEAD nonce (see {{multipath-aead}}). To ensure a unique nonce, path identifiers
+are limited to 32 bits and cannot be reused for another path of the same connection.
+However, note that with the use of this extension, multiple nonces can be in use simultaneously
+for the same AEAD key.
```

Comment:
This is a first shot, do not hesitate
```suggestion
This implies maintaining some per-path nonce computation state (a Path ID
and a packet number). Altogether, this ensures unique nonces over all the paths for
a same AEAD key.
```

Reply:
Should we even say:

"this implies maintaining per-path nonce values and computation state..."

?

Reply:
Okay, how about

"The multipath extension maintains unique per-path nonces for a same AEAD key."

That makes the state aspect a bit more clear...?

Reply:
```suggestion
The multipath extension maintains unique per-path nonces for a same AEAD key.
```
Is that okay @qdeconinck ?

Reply:
The only issue I have with this wording is that one may thing nonces are unique per path, which is not correct (nonces are computed in a way that two different paths will always have different values, hence guaranteeing their uniqueness over the whole connection). I fear this last version is ambiguous on this.

Reply:
Should we just remove the word "unique" then? "nonces for a same AEAD key" implies that they are unique, no?

Reply:
The method described in {{multipath-aead}} defines the AEAD nonce as a the combination of the IV, the 32 bit path ID, and the 62 bit packet sequence number. The path ID is unique to the path, and a given 62 bit sequence number is only use for a single packet on that path. Combining path ID and sequence number guarantees that AEAD will use a different nonce for each packet sent on the connection.

Reply:
yes, the point was it's not only unique for the path but actually unique for the whole connection. However, the fact they are unique is not really the point of this sentence, so that's why I proposed to just remove that word here.

Reply:
It boils down to the definition of "nonce". By definition, "nonce" means "only once", and thus "unique". What the AEAD specification says is a bit fuzzy, with nonce used to designate either the combination of IV+sequence number or the sequence number itself. My preference would be:

The method described in {{multipath-aead}} provides unique AEAD nonces for all packets across all path.